Fix : Path tracker causes app to crash #362 (made changes after reviewed)

reverie-ss opened the pull request:
- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

Fix : Path tracker causes app to crash #362

Problem : App used to crash on clicking "Start Tracking " button in PathTrackingActivity
Reason : Location permission and request was not handled
Solution : Added run-time location permission request in PatchTrackingActivity

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java:
```diff
@@ -90,4 +118,51 @@ public void onServiceDisconnected(ComponentName componentName) {
+
+        if (!gpsEnabled && !networkEnabled) {
+            // notify user
+            final AlertDialog.Builder builder = new AlertDialog.Builder(this);
```
Please use https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/mifosxdroid/core/MaterialDialog.java for showing material dialog.

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java:
```diff
@@ -40,19 +48,38 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
+                            "Permission Denied");
+        } else {
+
+            start.setOnClickListener(new View.OnClickListener() {
```
we are using ButterKnife So we do not need to implement ```View.OnClickListener()```
Use ```onClick``` annotation instead.

reverie-ss replied:
`start.setOnClickListener(new View.OnClickListener() {})`

It's there in the master code! I haven't added!

(https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java#L43)

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java:
```diff
@@ -90,4 +118,51 @@ public void onServiceDisconnected(ComponentName componentName) {
+        if (!gpsEnabled && !networkEnabled) {
+            // notify user
+            final AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setMessage("Location seems to be disabled, do you want to enable it?")
```
Do not user hard coded strings.

therajanmaurya commented:
@reverie-ss Please make above changes :)

reverie-ss commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java:
```diff
@@ -40,19 +48,38 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
-            }
-        });
+
+        if (!CheckSelfPermissionAndRequest
```
Followed the implementation!
And I have added `onRequestPermissionsResult`

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java:
```diff
@@ -41,27 +50,33 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
-                service.startLogging();
-            }
-        });
+                @Override
```
Remove this View.OnClickListener implementation and use ButterKnife annotation.

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java:
```diff
@@ -41,27 +50,33 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
-                service.stopLogging();
-            }
-        });
+                @Override
```
same as above

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java:
```diff
@@ -41,27 +50,33 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         showBackButton();
     }
 
+
```
do not add the blank extra line, remove this line.

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java:
```diff
@@ -41,27 +50,33 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         Intent intent = new Intent(this, LocationService.class);
         startService(intent);
         bindService(intent, this, Context.BIND_AUTO_CREATE);
+
```
same as above

therajanmaurya commented:
This PR needs some changes please look comments.
and Before sending final PR please squash all your commits into one.

reverie-ss commented:
-Fixed the issues in the comments
-Squashed all the commits into one

**I have made a new final PR request #418 . This PR can be closed.**
Thank you for your time.
